<?xml version="1.0" encoding="UTF-8"?>
<testng-results skipped="0" failed="3" total="8" passed="5">
  <reporter-output>
  </reporter-output>
  <suite name="Suite" duration-ms="7984" started-at="2019-05-04T20:44:53Z" finished-at="2019-05-04T20:45:01Z">
    <groups>
    </groups>
    <test name="Test" duration-ms="7984" started-at="2019-05-04T20:44:53Z" finished-at="2019-05-04T20:45:01Z">
      <class name="com.madhub.demo.codingTest.TC4_PostServiceTest">
        <test-method status="FAIL" signature="POSTServiceTest()[pri:0, instance:com.madhub.demo.codingTest.TC4_PostServiceTest@50a638b5]" name="POSTServiceTest" duration-ms="1508" started-at="2019-05-05T02:14:57Z" finished-at="2019-05-05T02:14:58Z">
          <exception class="junit.framework.AssertionFailedError">
            <message>
              <![CDATA[[Validating the resource created by POST request]: Resource not created. [Status Code Validation]: junit.framework.AssertionFailedError: expected:<200> but was:<404>]]>
            </message>
            <full-stacktrace>
              <![CDATA[junit.framework.AssertionFailedError: [Validating the resource created by POST request]: Resource not created. [Status Code Validation]: junit.framework.AssertionFailedError: expected:<200> but was:<404>
	at junit.framework.Assert.fail(Assert.java:50)
	at core.ResponseValidators.validateResourceCreated(ResponseValidators.java:46)
	at com.madhub.demo.codingTest.TC4_PostServiceTest.verifyRecordCreated(TC4_PostServiceTest.java:100)
	at com.madhub.demo.codingTest.TC4_PostServiceTest.POSTServiceTest(TC4_PostServiceTest.java:86)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:86)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:822)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1130)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:782)
	at org.testng.TestRunner.run(TestRunner.java:632)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:366)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:361)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:319)
	at org.testng.SuiteRunner.run(SuiteRunner.java:268)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1244)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
	at org.testng.TestNG.run(TestNG.java:1064)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:113)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:206)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:177)
]]>
            </full-stacktrace>
          </exception> <!-- junit.framework.AssertionFailedError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- POSTServiceTest -->
      </class> <!-- com.madhub.demo.codingTest.TC4_PostServiceTest -->
      <class name="com.madhub.demo.codingTest.TC1_GetServiceTest">
        <test-method status="PASS" signature="initialization()[pri:0, instance:com.madhub.demo.codingTest.TC1_GetServiceTest@636be97c]" name="initialization" is-config="true" duration-ms="375" started-at="2019-05-05T02:14:53Z" finished-at="2019-05-05T02:14:53Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- initialization -->
        <test-method status="PASS" signature="initialization()[pri:0, instance:com.madhub.demo.codingTest.TC4_PostServiceTest@50a638b5]" name="initialization" is-config="true" duration-ms="2" started-at="2019-05-05T02:14:53Z" finished-at="2019-05-05T02:14:53Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- initialization -->
        <test-method status="PASS" signature="initialization()[pri:0, instance:com.madhub.demo.codingTest.TC5_PutServiceTest@1817d444]" name="initialization" is-config="true" duration-ms="1" started-at="2019-05-05T02:14:53Z" finished-at="2019-05-05T02:14:53Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- initialization -->
        <test-method status="PASS" signature="initialization()[pri:0, instance:com.madhub.demo.codingTest.TC6_DeleteServiceTest@6ca8564a]" name="initialization" is-config="true" duration-ms="1" started-at="2019-05-05T02:14:53Z" finished-at="2019-05-05T02:14:53Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- initialization -->
        <test-method status="PASS" signature="initialization()[pri:0, instance:com.madhub.demo.codingTest.TC7_GetServiceLowerBoundaryCase@50b472aa]" name="initialization" is-config="true" duration-ms="2" started-at="2019-05-05T02:14:53Z" finished-at="2019-05-05T02:14:53Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- initialization -->
        <test-method status="PASS" signature="initialization()[pri:0, instance:com.madhub.demo.codingTest.TC8_GetServiceUpperBoundaryCase@31368b99]" name="initialization" is-config="true" duration-ms="1" started-at="2019-05-05T02:14:53Z" finished-at="2019-05-05T02:14:53Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- initialization -->
        <test-method status="PASS" signature="GETServiceTest()[pri:0, instance:com.madhub.demo.codingTest.TC1_GetServiceTest@636be97c]" name="GETServiceTest" duration-ms="2636" started-at="2019-05-05T02:14:53Z" finished-at="2019-05-05T02:14:56Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- GETServiceTest -->
      </class> <!-- com.madhub.demo.codingTest.TC1_GetServiceTest -->
      <class name="com.madhub.demo.codingTest.TC5_PutServiceTest">
        <test-method status="FAIL" signature="PUTServiceTest()[pri:0, instance:com.madhub.demo.codingTest.TC5_PutServiceTest@1817d444]" name="PUTServiceTest" duration-ms="1037" started-at="2019-05-05T02:14:58Z" finished-at="2019-05-05T02:14:59Z">
          <exception class="junit.framework.AssertionFailedError">
            <message>
              <![CDATA[[Validting Newly Created/Updated Resource]: "title" value is not as expected.  Value requested =abc, but received=sunt aut facere repellat provident occaecati excepturi optio reprehenderit]]>
            </message>
            <full-stacktrace>
              <![CDATA[junit.framework.AssertionFailedError: [Validting Newly Created/Updated Resource]: "title" value is not as expected.  Value requested =abc, but received=sunt aut facere repellat provident occaecati excepturi optio reprehenderit
	at junit.framework.Assert.fail(Assert.java:50)
	at core.ResponseValidators.compareResponseData(ResponseValidators.java:67)
	at com.madhub.demo.codingTest.TC5_PutServiceTest.PUTServiceTest(TC5_PutServiceTest.java:117)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:86)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:822)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1130)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:782)
	at org.testng.TestRunner.run(TestRunner.java:632)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:366)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:361)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:319)
	at org.testng.SuiteRunner.run(SuiteRunner.java:268)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1244)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
	at org.testng.TestNG.run(TestNG.java:1064)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:113)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:206)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:177)
]]>
            </full-stacktrace>
          </exception> <!-- junit.framework.AssertionFailedError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- PUTServiceTest -->
      </class> <!-- com.madhub.demo.codingTest.TC5_PutServiceTest -->
      <class name="com.madhub.demo.codingTest.TC2_GetServiceTestSingleRecord">
        <test-method status="PASS" signature="GETServiceTestSingleRecord()[pri:0, instance:com.madhub.demo.codingTest.TC2_GetServiceTestSingleRecord@1725dc0f]" name="GETServiceTestSingleRecord" duration-ms="203" started-at="2019-05-05T02:14:56Z" finished-at="2019-05-05T02:14:56Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- GETServiceTestSingleRecord -->
      </class> <!-- com.madhub.demo.codingTest.TC2_GetServiceTestSingleRecord -->
      <class name="com.madhub.demo.codingTest.TC7_GetServiceLowerBoundaryCase">
        <test-method status="PASS" signature="GetServiceLowerBoundaryCase()[pri:0, instance:com.madhub.demo.codingTest.TC7_GetServiceLowerBoundaryCase@50b472aa]" name="GetServiceLowerBoundaryCase" duration-ms="681" started-at="2019-05-05T02:15:00Z" finished-at="2019-05-05T02:15:01Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- GetServiceLowerBoundaryCase -->
      </class> <!-- com.madhub.demo.codingTest.TC7_GetServiceLowerBoundaryCase -->
      <class name="com.madhub.demo.codingTest.TC8_GetServiceUpperBoundaryCase">
        <test-method status="PASS" signature="GetServiceUpperBoundaryCase()[pri:0, instance:com.madhub.demo.codingTest.TC8_GetServiceUpperBoundaryCase@31368b99]" name="GetServiceUpperBoundaryCase" duration-ms="210" started-at="2019-05-05T02:15:01Z" finished-at="2019-05-05T02:15:01Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- GetServiceUpperBoundaryCase -->
      </class> <!-- com.madhub.demo.codingTest.TC8_GetServiceUpperBoundaryCase -->
      <class name="com.madhub.demo.codingTest.TC3_GetServiceInvalidResource">
        <test-method status="PASS" signature="GETServiceWithInvalidResource()[pri:0, instance:com.madhub.demo.codingTest.TC3_GetServiceInvalidResource@3911c2a7]" name="GETServiceWithInvalidResource" duration-ms="749" started-at="2019-05-05T02:14:56Z" finished-at="2019-05-05T02:14:57Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- GETServiceWithInvalidResource -->
      </class> <!-- com.madhub.demo.codingTest.TC3_GetServiceInvalidResource -->
      <class name="com.madhub.demo.codingTest.TC6_DeleteServiceTest">
        <test-method status="FAIL" signature="DeleteServiceTest()[pri:0, instance:com.madhub.demo.codingTest.TC6_DeleteServiceTest@6ca8564a]" name="DeleteServiceTest" duration-ms="543" started-at="2019-05-05T02:14:59Z" finished-at="2019-05-05T02:15:00Z">
          <exception class="junit.framework.AssertionFailedError">
            <message>
              <![CDATA[[Validating Json Schema]: Json Schema validation failed. 
1 expectation failed.
Response body doesn't match expectation.
Expected: The content to match the given JSON schema.
warning: the following keywords are unknown and will be ignored: [$id]
    level: "warning"
    schema: {"loadingURI":"file:/C:/Users/basavma/RestAssuredDemo/target/classes/JsonSchemaSingleRecord.json#","pointer":""}
    domain: "syntax"
    ignored: ["$id"]
warning: the following keywords are unknown and will be ignored: [$id, examples]
    level: "warning"
    schema: {"loadingURI":"file:/C:/Users/basavma/RestAssuredDemo/target/classes/JsonSchemaSingleRecord.json#","pointer":"/properties/body"}
    domain: "syntax"
    ignored: ["$id","examples"]
warning: the following keywords are unknown and will be ignored: [$id, examples]
    level: "warning"
    schema: {"loadingURI":"file:/C:/Users/basavma/RestAssuredDemo/target/classes/JsonSchemaSingleRecord.json#","pointer":"/properties/id"}
    domain: "syntax"
    ignored: ["$id","examples"]
warning: the following keywords are unknown and will be ignored: [$id, examples]
    level: "warning"
    schema: {"loadingURI":"file:/C:/Users/basavma/RestAssuredDemo/target/classes/JsonSchemaSingleRecord.json#","pointer":"/properties/title"}
    domain: "syntax"
    ignored: ["$id","examples"]
warning: the following keywords are unknown and will be ignored: [$id, examples]
    level: "warning"
    schema: {"loadingURI":"file:/C:/Users/basavma/RestAssuredDemo/target/classes/JsonSchemaSingleRecord.json#","pointer":"/properties/userId"}
    domain: "syntax"
    ignored: ["$id","examples"]
error: object has missing required properties (["body","id","title","userId"])
    level: "error"
    schema: {"loadingURI":"file:/C:/Users/basavma/RestAssuredDemo/target/classes/JsonSchemaSingleRecord.json#","pointer":""}
    instance: {"pointer":""}
    domain: "validation"
    keyword: "required"
    required: ["body","id","title","userId"]
    missing: ["body","id","title","userId"]

  Actual: {}
]]>
            </message>
            <full-stacktrace>
              <![CDATA[junit.framework.AssertionFailedError: [Validating Json Schema]: Json Schema validation failed. 
1 expectation failed.
Response body doesn't match expectation.
Expected: The content to match the given JSON schema.
warning: the following keywords are unknown and will be ignored: [$id]
    level: "warning"
    schema: {"loadingURI":"file:/C:/Users/basavma/RestAssuredDemo/target/classes/JsonSchemaSingleRecord.json#","pointer":""}
    domain: "syntax"
    ignored: ["$id"]
warning: the following keywords are unknown and will be ignored: [$id, examples]
    level: "warning"
    schema: {"loadingURI":"file:/C:/Users/basavma/RestAssuredDemo/target/classes/JsonSchemaSingleRecord.json#","pointer":"/properties/body"}
    domain: "syntax"
    ignored: ["$id","examples"]
warning: the following keywords are unknown and will be ignored: [$id, examples]
    level: "warning"
    schema: {"loadingURI":"file:/C:/Users/basavma/RestAssuredDemo/target/classes/JsonSchemaSingleRecord.json#","pointer":"/properties/id"}
    domain: "syntax"
    ignored: ["$id","examples"]
warning: the following keywords are unknown and will be ignored: [$id, examples]
    level: "warning"
    schema: {"loadingURI":"file:/C:/Users/basavma/RestAssuredDemo/target/classes/JsonSchemaSingleRecord.json#","pointer":"/properties/title"}
    domain: "syntax"
    ignored: ["$id","examples"]
warning: the following keywords are unknown and will be ignored: [$id, examples]
    level: "warning"
    schema: {"loadingURI":"file:/C:/Users/basavma/RestAssuredDemo/target/classes/JsonSchemaSingleRecord.json#","pointer":"/properties/userId"}
    domain: "syntax"
    ignored: ["$id","examples"]
error: object has missing required properties (["body","id","title","userId"])
    level: "error"
    schema: {"loadingURI":"file:/C:/Users/basavma/RestAssuredDemo/target/classes/JsonSchemaSingleRecord.json#","pointer":""}
    instance: {"pointer":""}
    domain: "validation"
    keyword: "required"
    required: ["body","id","title","userId"]
    missing: ["body","id","title","userId"]

  Actual: {}

	at junit.framework.Assert.fail(Assert.java:50)
	at core.ResponseValidators.validateGetResponse(ResponseValidators.java:127)
	at com.madhub.demo.codingTest.TC6_DeleteServiceTest.DeleteServiceTest(TC6_DeleteServiceTest.java:68)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:86)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:822)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1130)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:782)
	at org.testng.TestRunner.run(TestRunner.java:632)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:366)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:361)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:319)
	at org.testng.SuiteRunner.run(SuiteRunner.java:268)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1244)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
	at org.testng.TestNG.run(TestNG.java:1064)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:113)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:206)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:177)
]]>
            </full-stacktrace>
          </exception> <!-- junit.framework.AssertionFailedError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- DeleteServiceTest -->
      </class> <!-- com.madhub.demo.codingTest.TC6_DeleteServiceTest -->
    </test> <!-- Test -->
  </suite> <!-- Suite -->
</testng-results>
